src/System.Composition.Convention/src/System/Composition/Convention/ConventionBuilder.cs

```diff
@@ -368,7 +368,7 @@ private static bool IsDescendentOf(Type type, Type baseType)
             var ti = type.GetTypeInfo();
             var bti = baseType.GetTypeInfo();
 
-            if (ti.IsGenericTypeDefinition)
+            if (ti.IsGenericTypeDefinition || bti.IsGenericTypeDefinition)
```

Comment:
Would you mind adding a comment explaining why we are now checking the baseType as well?

It will help in the future when someone else looks at this code.